pkg/controller/certificates/signer/config/types.go

```diff
@@ -28,7 +28,26 @@ type CSRSigningControllerConfiguration struct {
 	// clusterSigningCertFile is the filename containing a PEM-encoded
 	// RSA or ECDSA private key used to issue cluster-scoped certificates
 	ClusterSigningKeyFile string
+
+	// kubeletServingSignerConfiguration holds the certificate and key used to issue certificates for the kubernetes.io/kubelet-serving signer
+	KubeletServingSignerConfiguration CSRSigningConfiguration
+	// kubeletClientSignerConfiguration holds the certificate and key used to issue certificates for the kubernetes.io/kube-apiserver-client-kubelet
+	KubeletClientSignerConfiguration CSRSigningConfiguration
```

Comment:
Do we intend/want to support having a distinct CA for kube-apiserver-client and kube-apiserver-client-kubelet? Presumably these would be reasonably expected to be two intermediaries from a common root (once there is better support for intermediaries too).

Unless it’s possible to specify multiple root CAs used to validate apiserver client certificates?

Reply:
> Unless it’s possible to specify multiple root CAs used to validate apiserver client certificates?

It is.  In fact openshift does this today.



> Do we intend/want to support having a distinct CA for kube-apiserver-client and kube-apiserver-client-kubelet?

Yes.  The kubelets have well defined and well behaved rotation capabilities that are managed, not external.  These properties mean that having comparatively short lived signers is possible.  In general, apiserver clients don't have these properties.